tests/controllers/globalroles/globalrole_test.go

```diff
@@ -0,0 +1,236 @@
+)
+
+func (s *GlobalRoleTestSuite) SetupSuite() {
+	s.ctx, s.cancel = context.WithCancel(context.TODO())
+
+	// Start envtest
+	s.testEnv = &envtest.Environment{}
+	restCfg, err := s.testEnv.Start()
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), restCfg)
+
+	// Register CRDs
+	common.RegisterCRDs(s.T(), s.ctx, restCfg,
+		crd.CRD{
+			SchemaObject: v3.GlobalRole{},
+			NonNamespace: true,
+			Status:       true,
+		},
+		crd.CRD{
+			SchemaObject: v3.GlobalRoleBinding{},
+			NonNamespace: true,
+		})
+
+	// Create wrangler context
+	wranglerContext, err := wrangler.NewContext(s.ctx, nil, restCfg)
+	assert.NoError(s.T(), err)
+
+	// Create management context
+	scaledContext, clusterManager, _, err := multiclustermanager.BuildScaledContext(s.ctx, wranglerContext, &multiclustermanager.Options{})
+	assert.NoError(s.T(), err)
+	s.managementContext, err = scaledContext.NewManagementContext()
+	assert.NoError(s.T(), err)
+
+	// Register controller
+	globalroles.Register(s.ctx, s.managementContext, clusterManager)
+
+	// Start controllers
+	common.StartNormanControllers(s.T(), s.ctx, s.managementContext,
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "GlobalRoleBinding",
+		},
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "GlobalRole",
+		})
+
+	// Start caches
+	common.StartWranglerCaches(s.T(), s.ctx, s.managementContext.Wrangler,
+		schema.GroupVersionKind{
+			Group:   "rbac.authorization.k8s.io",
+			Version: "v1",
+			Kind:    "ClusterRole",
+		},
+		schema.GroupVersionKind{
+			Group:   "rbac.authorization.k8s.io",
+			Version: "v1",
+			Kind:    "ClusterRoleBinding",
+		},
+		schema.GroupVersionKind{
+			Group:   "rbac.authorization.k8s.io",
+			Version: "v1",
+			Kind:    "Role",
+		},
+		schema.GroupVersionKind{
+			Group:   "rbac.authorization.k8s.io",
+			Version: "v1",
+			Kind:    "RoleBinding",
+		},
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "GlobalRole",
+		})
+}
+
+func (s *GlobalRoleTestSuite) TearDownSuite() {
+	s.cancel()
+	err := s.testEnv.Stop()
+	assert.NoError(s.T(), err)
+}
+
+func (s *GlobalRoleTestSuite) TestCreateGlobalRole() {
+	tests := []struct {
+		name        string
+		globalRole  v3.GlobalRole
+		roles       []rbacv1.Role
+		clusterRole rbacv1.ClusterRole
+	}{
+		{
+			name: "create primary cluster role given cr-name",
+			globalRole: v3.GlobalRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						crNameLabel: "cr-name",
+					},
+					Name: "test-gr",
+				},
+				Rules: []rbacv1.PolicyRule{getPodRule},
+			},
+			clusterRole: rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cr-name",
+				},
+				Rules: []rbacv1.PolicyRule{getPodRule},
+			},
+		},
+		{
+			name: "create primary cluster role with generated name",
+			globalRole: v3.GlobalRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-gr",
```

Comment:
I would use a different name to avoid conflics with the previous test. I got an error saying that the GlobalRole existed because it was still deleting the previous one.